selfservice/strategy/oidc/strategy.go

```diff
@@ -169,8 +173,13 @@
 				}
 
 				for _, prov := range conf.Providers {
-					if provider == prov.Provider && sub == prov.Subject &&
-						prov.Subject != "" && prov.Provider != "" {
+					if withOrgs && len(prov.Organization) == 0 {
```

Comment:
Shouldn't we just count orgs that the identity is currently in here?

Reply:
This specific OIDC strategy cannot deal with organizations at all - it is only for social sign in. So the correct way is to exclude any organization providers as that would be handled by a different strategy. I can link you to the code in the private repo for clarification if you want!

Reply:
I think this is conceptually sound and we can still fix this up if not :)